Connect to the Movistar SMS API

Senen opened the pull request:
## References
Related Issues: #5 

## Objectives
Connect to the [API Mensajería Negocios v3.0](https://www.mensajerianegocios.movistar.es/Manual_Mensajeria_Negocios_API.pdf)

## Notes
This PR adds a new secret entry so remember to update the `secrets.yml` file on your servers with the `Sender` code. Movistar will provide your account with this identifier.

**Testing successfully on production environment!**

javierm commented on lib/sms_api.rb:
```diff
@@ -19,19 +19,23 @@ def authorization
+      DeliveryReport: "All",
+      Recipients: { to: phone },
+      Sender: Rails.application.secrets.sms_sender,
+      SMSText: "Clave para verificarte: #{code}. Gobierno Abierto",
```
Would it be possible to internationalize this text? So users browsing the web in Galician receive a message written in Galician.

Senen replied:
Done

javierm commented on lib/sms_api.rb:
```diff
@@ -6,6 +6,15 @@ def initialize
+    success?(response)
+  end
+
+  private
```
While I think these changes are useful, I'd rather have the code as similar to CONSUL's code as possible so it's easier to deal with conflicts when upgrading :thinking:. But if you think differently, we can include this change :wink:.

Senen replied:
Done

javierm commented on lib/sms_api.rb:
```diff
@@ -19,19 +19,23 @@ def authorization
-      solicita_notificacion: "All" }
+    {
+      Authorization: authorization,
+      DeliveryReport: "All",
```
Since we're using `convert_request_keys_to: :camelcase`, do we have to use camel case here?

javierm commented on spec/lib/sms_api_spec.rb:
```diff
@@ -7,5 +8,37 @@
       expect(api.sms_deliver("666777888", "AAAA")).to be_falsey
     end
+
+    context "#when endpoint is defined" do
```
Typo: the  "#" is not necessary.

javierm reviewed (commented):
Nice investigation. Let's see whether it works or we need to polish the format.

As usual, I've left a couple of comments :wink:.

Senen committed:
Update SMSApi lib to connect with Movistar

Update SMSApi lib to connect with Movistar Mensajería de Negocios API.

Add SMS message translations for enabled languages.
